endpoint.go

```diff
@@ -456,15 +456,23 @@ func (ep *endpoint) sbJoin(sb *sandbox, options ...EndpointOption) error {
 		if extEp != nil {
 			log.Debugf("Revoking external connectivity on endpoint %s (%s)", extEp.Name(), extEp.ID())
 			if err := d.RevokeExternalConnectivity(extEp.network.ID(), extEp.ID()); err != nil {
-				log.Warnf("driver failed revoking external connectivity on endpoint %s (%s): %v",
+				return types.InternalErrorf(
+					"driver failed revoking external connectivity on endpoint %s (%s): %v",
 					extEp.Name(), extEp.ID(), err)
 			}
+			defer func() {
+				if err := d.ProgramExternalConnectivity(extEp.network.ID(), extEp.ID(), sb.Labels()); err != nil {
```

Comment:
The unconditional call is a bug. Thanks for catching it.
The clear default gw has always been a best effort, we have not rolled back a join if that fails.
IMO it should still behave that way,.